### PR TITLE
Enable/Disable by glob

### PR DESCRIPTION
**Pull request by yosiat**
Issue - #337 

- Add FindTasks method for fetching all rawTask[] by a predicate
- Enabling/Disabling by glob

For example:
`> kapacitor enable "window_*"` 
Will enable all tasks that starts with "window_"
The same for disable

**Notes:**
- I think this would be nice for delete as well
- The enable is not efficient as possible - I am accessing boltdb for each task, I can change it, but I think it will hurt the readability
- If at least one task is failed to enable/disable I am stopping and returning an error - are we ok about this behaviour?

### CODE REVIEW

**Comment by nathanielc** [comment]
> I think this would be nice for delete as well

+1

>The enable is not efficient as possible - I am accessing boltdb for each task, I can change it, but I think it will hurt the readability

Simple is better

> If at least one task is failed to enable/disable I am stopping and returning an error - are we ok about this behavior?

Yes

File: services/task_store/service.go [hunk 1]
```diff
@@ -706,56 +762,95 @@ func (ts *Service) IsEnabled(name string) (e bool) {
...
+			return true, nil
+		}
+
+		// This code is not the optimal code ever, we could use map here
+		// but I am assuming the asked tasks count will be small
+		for _, askedTaskName := range tasks {
+			if askedTaskName == taskName {
+				return true, nil
+			}
+		}
```
Comment (nathanielc) [inline services/task_store/service.go]: If we are not doing any matching here why are we not looping over `LoadRaw`?

Comment (yosiat) [inline services/task_store/service.go]: Umm.. I agree, but the reason we need to handle the case where tasks is empty, and we want to fetch all tasks.

The looping LoadRaw is more simple & elegant.. do you have any way can do this and still support the empty tasks array?

Comment (nathanielc) [inline services/task_store/service.go]: I may be missing it but I do not see the change in behavior from the previous implementation to this one?

Comment (yosiat) [inline services/task_store/service.go]: Let's make sure we are on the same idea, you are suggesting that instead of using FindTasks and looking at the bucket we will do something like this:
```
for taskName := range tasks {
   raw:=ts.LoadRaw(taskName)
  // convert to TaskSummaryInfo, and IsEnabled, and ExecutionStatus
}
```
If you are suggesting doing this approach, we have the case where the tasks are empty and we want to fetch all tasks - "/tasks",  so we are using FindTasks to find the matching tasks.

Comment (nathanielc) [inline services/task_store/service.go]: I am not suggesting that, but rather looking at the original implementation it handled the case where tasks was empty via the `ForEach` boltdb bucket call. I see that method has changed but I do not see how its behavior has changed so I am confused as to why it was modified. I am probably just missing something....

Comment (yosiat) [inline services/task_store/service.go]: ok, you are suggesting to undo the change in this function? I changed this function to use the FindTasks, but after a little bit of thinking it makes things harder.

Comment (nathanielc) [inline services/task_store/service.go]: Yes, there is no need to use the FindTasks if it is not necessary.

**Comment by yosiat** [comment]
@nathanielc are we ok on this PR or do you want me to implement delete by glob as well before we merge?

**Comment by yosiat** [comment]
Added delete by glob

**Comment by nathanielc** [comment]
@yosiat This looks good to go. Do you want to add a CHANGELOG entry real quick?

**Comment by yosiat** [comment]
I will do in one hour, I forgot to update the usage for the relevant
commands in kapacitor cmd

**Comment by yosiat** [comment]
@nathanielc Updated and added help usage..
please make sure it's ok, and I will squash the commits

File: cmd/kapacitor/main.go [hunk 2]
```diff
@@ -874,7 +904,24 @@ func deleteUsage() {
...
+
+    $ kapacitor delete tasks *_alert 
+
+		You can delete recordings:
+
+    $ kapacitor delete recordings b0a2ba8a-aeeb-45ec-bef9-1a2939963586 
+
+		Or, you can enable by glob:
+
+    $ kapacitor enable *_alert
```
Comment (nathanielc) [inline cmd/kapacitor/main.go]: These lines look like there are extras?

Comment (yosiat) [inline cmd/kapacitor/main.go]: Oops.. I did copy/paste, Will fix this.

**Comment by nathanielc** [comment]
@yosiat Looks good, I'll merge when its squashed

**Comment by yosiat** [comment]
@nathanielc squashed

**Comment by fatihcode** [comment]
@yosiat : does it mean `kapacitor enable "*"` and `kapacitor disable "*"` enabled and disable all the tasks?

**Comment by yosiat** [comment]
@fatihcode yes
